Sources/OutlineView/OutlineViewDelegate.swift

```diff
@@ -102,27 +102,36 @@ where Data.Element: Identifiable {
     func outlineViewItemDidExpand(_ notification: Notification) {
         let outlineView = notification.object as! NSOutlineView
         if outlineView.selectedRow == -1 {
-            selectRow(for: selectedItem, in: outlineView)
+            selectRow(for: selectedItems, in: outlineView)
         }
     }
 
     func outlineViewSelectionDidChange(_ notification: Notification) {
         let outlineView = notification.object as! NSOutlineView
-        if outlineView.selectedRow != -1 {
-            let newSelection = outlineView.item(atRow: outlineView.selectedRow).map(typedItem)
-            if selectedItem?.id != newSelection?.id {
-                selectedItem = newSelection
-                selectionChanged(selectedItem?.value)
+        let selectedRowIndexes = outlineView.selectedRowIndexes
+        
+        if !selectedRowIndexes.isEmpty {
+            let newSelections = selectedRowIndexes.compactMap {
+                outlineView.item(atRow: $0).map(typedItem)
+            }
+            
+            if selectedItems.count == newSelections.count {
+                return
+            }
+                       
+            if selectedItems.allSatisfy({ newSelections.contains($0) }) {
+                selectedItems = newSelections
+                selectionChanged(Set(selectedItems.map(\.value)))
             }
         }
     }
 
     func selectRow(
-        for item: OutlineViewItem<Data>?,
+        for items: [OutlineViewItem<Data>],
         in outlineView: NSOutlineView
     ) {
         // Returns -1 if row is not found.
-        let index = outlineView.row(forItem: selectedItem)
+        let index = outlineView.row(forItem: selectedItems)
```

Comment:
I don't think the updated code is correct for multiple selected items. I grabbed your code for an app I'm working on and besides renaming it selectRows my current code is:
func selectRows(
        for items: [OutlineViewItem<Data>],
        in outlineView: NSOutlineView
    ) {
        // Returns -1 if row is not found.
        let indices = items.compactMap { outlineView.row(forItem: $0) == -1 ? nil : outlineView.row(forItem: $0) }
        outlineView.selectRowIndexes(IndexSet(indices), byExtendingSelection: false)
    }